### DISCUSSION TITLE
Welcome to xml_tv Discussions!

### DISCUSSION THREAD

**Discussion by markcs** [markcs/xml_tv#38] [General]
## 👋 Welcome!
  I thought I'd enable Github Discussions as a place where xml_tv users can connect. I hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with and help other community members.

**Comment by texass44** [comment]
Hi Mark......love the work you've done here.....brilliant !!
I've actually been using the EPG data for Melbourne for a long time now in Digital Now's "DNTV Scheduler Pro 2" product and it works really well.  Its long in the tooth and starting to show some cracks so I've just updated my HTPC to Win10 and decided to try out "TV Scheduler Pro" as my DTV recording solution.  Now when I load the Melbourne data into TV Scheduler Pro I get an EPG that spans many months, back to 2020.  I cant find a way in TV Scheduler Pro to limit how far back EPG data is displayed for.  
In the XML download I can find the following very old entries

<programme start="20200724070000 +0000" stop="20200724090000 +0000" channel="202.yourtv.com.au">
<title>Good Morning</title>
<desc>
Music for early risers featuring Play School, Giggle and Hoot, The Wiggles and heaps more.
</desc>
<category>Radio</category>
</programme>

and 

<programme start="20210422230000 +0000" stop="20210423020000 +0000" channel="201.yourtv.com.au">
<title>Arvos On Jazz</title>
<desc>
Keep the day rolling with the best jazz from Australia and abroad, including another Jazz Track from Mal Stanley (4pm) and a segment from our Artist in Residence (2pm).
</desc>
<category>Radio</category>
</programme>

Would it be possible to remove these old entries ??

Thanks & keep up the great work, it is VERY much appreciated !!!

Ken.

**Reply by texass44** [reply]
Sorry....looks like the xml I pasted got a bit smashed....I'll try again....

-programme start="20200724070000 +0000" stop="20200724090000 +0000" channel="202.yourtv.com.au">
-title>Good Morning</title>
-desc>
Music for early risers featuring Play School, Giggle and Hoot, The Wiggles and heaps more.
-/desc>
-category>Radio</category>
-/programme

and

-programme start="20210422230000 +0000" stop="20210423020000 +0000" channel="201.yourtv.com.au">
-title>Arvos On Jazz</title>
-desc>
Keep the day rolling with the best jazz from Australia and abroad, including another Jazz Track from Mal Stanley (4pm) and a segment from our Artist in Residence (2pm).
-/desc>
-category>Radio</category>
-/programme

**Reply by markcs** [reply]
Hi ken,

Sorry, I'm a bit confused here!  

Are you running the yourtv.pl script locally yourself or using the data from xmltv.net?  

Either way, each time that script is run or data is downloaded, it will only cover at most 7 days.  I've never used TV Scheduler Pro or DNTV Scheduler Pro, but are those programs caching old data somehow?

**Comment by texass44** [comment]
Hi Mark…..wow that’s a super fast response.

I’m just downloading the XML via the programs. If you download the XML in a
browser and look for “2020 you’ll see the first old entry.  Look for
“202104 will show the other old entry.

Thanks Ken.

**Reply by markcs** [reply]
So downloading the file from http://xmltv.net/xml_files/Melbourne.xml  ?

**Reply by texass44** [reply]
Yes.....thats the one.  That XML looks like its got those 2 old entries in there ....

**Reply by markcs** [reply]
Hmm, you're right.  I wonder where those come from.  
Both of those are for the radio station.  Let me check later what is happening.

```
        <programme start="20210422230000 +0000" stop="20210423020000 +0000" channel="201.yourtv.com.au">
                <title>Arvos On Jazz</title>
                <desc>Keep the day rolling with the best jazz from Australia and abroad, including another Jazz Track from Mal Stanley (4pm) and a segment from our Artist in Residence (2pm).</desc>
                <category>Radio</category>
        </programme>
```

**Reply by texass44** [reply]
Thanks Mark.......that would be awesome !!!

Again, thanks for all your hard work !!

Cheers.....Ken.

**Reply by markcs** [reply]
Fixed!  
I'm trying to get hold of the guy that runs xmltv.net to update the script on his server. Hopefully I'll get him to do that soon ...

**Comment by texass44** [comment]
Great work Mark.....thank you so much.

Cheers Ken.

**Comment by texass44** [comment]
Hi Mark....thanks, I can see some old entries gone now......great !!

However, sorry to be a pain but there is a few more lurking about.....

(programme start="20210422230000 +0000" stop="20210423020000 +0000"            
channel="201.yourtv.com.au")                                                   
(title)Arvos On Jazz(/title)                                                   
(desc)                                                                         
Keep the day rolling with the best jazz from Australia and abroad, including another Jazz Track from Mal Stanley (4pm) and a segment from our Artist in Residence (2pm).                                                               
(/desc)                                                                        
(category)Radio(/category)                                                     
(/programme)                                    

and 

(programme start="20210501080000 +0000" stop="20210501140000 +0000"          
channel="201.yourtv.com.au")                                                 
(title)Jazz After Hours(/title)                                              
(desc)                                                                       
Accompanying nocturnal and hibernating Jazz Cats around Australia. including 
(/desc) Jazz Track from Mal Stanley (4pm) and a segment from our Artist in   
(category)Radio(/category)                                                   
(/programme)                                                                                                

Cheers & best regards.......Ken.

**Reply by markcs** [reply]
The script had not been updated on xmltv.net yet. I'm not sure when (I can't get hold of the guy managing it), but hopefully soon

**Comment by texass44** [comment]
no probs.....thanks

**Reply by markcs** [reply]
Hopefully it is updated now. Can you confirm?

**Comment by texass44** [comment]
Hi Mark.....looks great.  Thank you so much.

Cheers !!    Ken

**Comment by oismacca** [comment]
Hi,
Is the data for WA working ok? I'm missing chunks of data about 1 day's worth but I'm not sure if its the way I grab/modify the data or not

**Reply by markcs** [reply]
There was a small change in one of the APIs, which may have been causing the issue.

Can you recheck and let me know?

**Reply by oismacca** [reply]
Cheers for the reply, but it ended up being an issue at my end.

**Comment by markcs** [comment]
@texass44  @oismacca  @markallen4123 

Not sure if you are still using the files from xmltv.net, but if you are, can I ask a favour?  Can you post your region name and a list of your channel names and channel numbers?  

We've found a few duplicated channels numbers and want to ensure we filter our the correct channels.  The duplicated channel numbers is no impact if Plex is already setup and working but it could be less usable for some people.  

Thanks!

**Reply by oismacca** [reply]
Yeah mate I am quite involved in collating info about FTA lineups in all markets so can definitely assist here. There are heaps of differences out there

**Metro (Sydney, Melbourne, Brisbane, Adelaide, Perth):**
1 & 15 = 10 HD
2 & 21 = ABC TV (SD)
7 & 71 = 7 (SD)
9 & 91 = Channel 9 (SD)
93 & 99 = 9Go!

**NSW (Albury, Orange, South Coast, Wagga Wagga, Wollongong):**
2 & 21 = ABC TV (SD)
5 & 51 = 10 (SD)
6 & 61 = 7 (SD)

**NSW (Central Coast, Coffs Harbour, Lismore, Newcastle, Tamworth, Taree/Port Macquarie):**
2 & 21 = ABC TV (SD)
6 & 61 = 7 (SD)
8 & 81 = Channel 9 (SD)
83 & 88 = 9Go!

**NSW (Griffith):**
2 & 21 = ABC TV (SD)
6 & 60 = 7HD
5 & 50 = 10 HD

**Victoria (Albury/Wodonga, Ballarat, Bendigo, Geelong, Shepparton):**
2 & 21 = ABC TV (SD)
5 & 51 = 10 (SD)
6 & 61 = 7 (SD)

**Victoria (Mildura)**
2 & 21 = ABC TV (SD)
5 & 50 = 10 HD
6 & 61 = 7 (SD)

**Queensland (Cairns, Mackay, Rockhampton, Toowoomba, Townsville, Wide Bay):**
2 & 21 = ABC TV (SD)
5 & 51 = 10 (SD)
7, 70 & 71 = 7HD

**Queensland (Sunshine Coast):**
2 & 21 = ABC TV (SD)
5 & 51 = 10 (SD)
6, 60 & 61 = 7HD

**Tasmania (Hobart, Launceston):**
2 & 21 = ABC TV (SD)
5 & 51 = 10 (SD)
6 & 61 = 7 (SD)

**South Australia (Riverland, South East SA):**
2 & 21 = ABC TV (SD)
5 & 50 = 10 HD
6 & 60 = 7HD

**South Australia (Spencer Gulf, Broken Hill):**
2 & 21 = ABC TV (SD)
5 & 51 = 10 (SD)
6 & 61 = 7 (SD)
8 & 81 = 9 (SD)

**Northern Territory (Darwin)**
1 & 11 = 10 (SD)
2 & 21 = ABC TV (SD)
7 & 71 = 7 (SD)
9 & 91 = Channel 9 (SD)

**Northern Territory (Regional NT)**
2 & 21 = ABC TV (SD)
9 & 91 = Imparja (SD)

**Western Australia (Albany, Bunbury, Jurien Bay, Mandurah, WA Regional)**
2 & 21 = ABC TV (SD)

**Reply by markcs** [reply]
Excellent info. Thanks @oismacca
